implementations/rust/ockam/ockam_abac/src/eval.rs

```diff
@@ -52,7 +52,13 @@ pub fn eval(expr: &Expr, env: &Env) -> Result<Expr, EvalError> {
                             if nargs != 3 {
                                 return Err(EvalError::malformed("'if' requires three arguments"))
                             }
-                            ctrl.push(Op::If)
+                            // We first evaluate the test and only then, depending on the result
+                            // do we either evaluate the true branch or the false branch.
+                            ctrl.push(Op::Eval(&xs[3])); // false branch
+                            ctrl.push(Op::Eval(&xs[2])); // true branch
+                            ctrl.push(Op::If);
+                            ctrl.push(Op::Eval(&xs[1])); // test
+                            continue
```

Comment:
Is there a test evaluating an expression like this as an example? Or at least, I think it would be great to have an example in the comment (line 56) to help visualize the `xs`'s expected elements.